src/server/transaction.cc

```diff
@@ -176,11 +176,17 @@ Transaction::~Transaction() {
 
 void Transaction::InitBase(Namespace* ns, DbIndex dbid, CmdArgList args) {
   global_ = false;
-  namespace_ = ns;
   db_index_ = dbid;
   full_args_ = args;
   local_result_ = OpStatus::OK;
   stats_.coordinator_index = ProactorBase::me() ? ProactorBase::me()->GetPoolIndex() : kInvalidSid;
+
+  // Namespace is read by poll execution, so it can't be changed on the fly
+  if (IsScheduled()) {
```

Comment:
Is it possible to check if ns_ is the same as ns when ns_ is not null?

Reply:
i.e. `if (ns_ != nullptr)  { DCHECK(ns_ == ns); }`

Reply:
Maybe `DHECK(ns_ == nullptr || ns_ == ns);`